### PR TITLE
Update nav item

### PR DESCRIPTION
**Pull request by jacobp100**
Rewrite of #10880: a pull request could not be merged due to CLA. This uses a different method, so there shouldn’t be any licensing issues.

### CODE REVIEW

**Comment by mkonicek** [comment]
Thanks @jacobp100! My understanding is #10880 didn't have licensing issues. The author just never got back to sign the CLA.

**Comment by jacobp100** [comment]
I meant I can’t copy his code and sign the CLA on his behalf!

**Comment by norman-kapschefsky** [comment]
@jacobp100 Sorry for my delayed reaction on my PR. I created a new one [#11129](https://github.com/facebook/react-native/pull/11129) after signing the CLA and I hope everything is fine now. I don't know which solution is better but I know that I want to have that fix/feature to be implemented ;)

**Comment by jacobp100** [comment]
Yours is better!

**Comment by jacobp100** [comment]
@mkonicek Looks like we didn’t get anywhere with the CLA. 😟 Would it be possible to merge this in instead? I’ll be able to make changes if needed.

**Comment by ericvicenti** [comment]
Can you update the description of the PR to be more descriptive?

Why is this change needed? Could it possibly break anything? How have you tested this? Are there additional APIs introduced? If so, they should be documented in this PR.

**Comment by jacobp100** [comment]
This is a reimplementation of #10880

> This pull request will fix the problem that the NavigatorIOS wont adjust its navigation bar when replacing the current route with new navigation bar properties.

> Within a component that is rendered by the NavigatorIOS the following lines of code should update the navigation bar (e.g. with a new title) but it won't.

> // somewhere within a react component
> var route = this.props.navigator.navigationContext.currentRoute;
> route.title = 'newTitle';
> this.props.navigator.replace(route);

Doesn't break anything, no new APIs, can be tested with above code. We could potentially add a UIExplorer example, but that might be overkill.

File: React/Views/RCTWrapperViewController.m [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #import "RCTUtils.h"
 #import "RCTViewControllerProtocol.h"
 #import "UIView+React.h"
-#import "RCTAutoInsetsProtocol.h"
+#import "RCTAutoInsetsProtocol.h"`
```
Comment (ericvicenti) [inline React/Views/RCTWrapperViewController.m]: Is this a typo? Does it cause any troubles?

Comment (jacobp100) [inline React/Views/RCTWrapperViewController.m]: Good spot—that was a typo. Not sure how that got past CI 😖

**Comment by jacobp100** [comment]
I fixed the typo. Is there anything more I need to do?

**Pull request closed without merging**